### #3 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### #4 and #5

In [2]:
gdp_df=pd.read_csv("C:/Users/Andrew/Documents/NSS/python/projects/un-project-Main-Im/data/gdp_percapita.csv")
gdp_df
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8219 entries, 0 to 8218
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  8219 non-null   object 
 1   Year             8219 non-null   int64  
 2   Value            8219 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 257.0+ KB


There are 8219 rows and 4 columns. Country or Area column type is object. Year column is int64. Value is float64. Value Footnotes is float64.

### #6

In [3]:
gdp_df=gdp_df.drop(columns='Value Footnotes')
gdp_df=gdp_df.rename(columns={'Country or Area':'Country','Year':'Year','Value':'GDP_per_Capita'})

### #7

In [4]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8219 entries, 0 to 8218
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         8219 non-null   object 
 1   Year            8219 non-null   int64  
 2   GDP_per_Capita  8219 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 192.8+ KB


In [6]:
gdp_df['Year'].min()

1990

In [13]:
gdp_df.Year.value_counts().to_frame().reset_index()

Year  count
0   2020    247
1   2021    247
2   2013    247
3   2016    247
4   2019    247
5   2018    247
6   2017    247
7   2014    247
8   2015    247
9   2022    246
10  2011    246
11  2012    246
12  2009    245
13  2010    245
14  2008    244
15  2007    242
16  2006    242
17  2005    241
18  2003    241
19  2004    241
20  2002    241
21  2023    240
22  2001    240
23  2000    240
24  1999    238
25  1998    238
26  1997    238
27  1996    236
28  1995    236
29  1994    235
30  1993    234
31  1992    234
32  1991    234
33  1990    233

In [10]:
gdp_df.Year.nunique()

34

In [7]:
gdp_df['Year'].max()

2023

In [8]:
gdp_df_2=gdp_df[['Country','Year']]
gdp_df_2

Country  Year
0     Afghanistan  2023
1     Afghanistan  2022
2     Afghanistan  2021
3     Afghanistan  2020
4     Afghanistan  2019
...           ...   ...
8214     Zimbabwe  1994
8215     Zimbabwe  1993
8216     Zimbabwe  1992
8217     Zimbabwe  1991
8218     Zimbabwe  1990

[8219 rows x 2 columns]

np.int64(8219)